File: internal/content/testdata/bench/java/nc-rotting-oranges.java
import java.util.*;

class Solution {
    public long oranges_rotting(long[][] grid) {
        int m = grid.length, n = grid[0].length;
        Queue<int[]> queue = new LinkedList<>();
        long fresh = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] == 2) queue.add(new int[]{i, j, 0});
                else if (grid[i][j] == 1) fresh++;
            }
        }
        if (fresh == 0) return 0;
        long minutes = 0;
        int[][] dirs = {{1,0},{-1,0},{0,1},{0,-1}};
        while (!queue.isEmpty()) {
            int[] cur = queue.poll();
            int i = cur[0], j = cur[1], t = cur[2];
            for (int[] d : dirs) {
                int ni = i + d[0], nj = j + d[1];
                if (ni >= 0 && ni < m && nj >= 0 && nj < n && grid[ni][nj] == 1) {
                    grid[ni][nj] = 2;
                    fresh--;
                    minutes = t + 1;
                    queue.add(new int[]{ni, nj, t+1});
                }
            }
        }
        return fresh == 0 ? minutes : -1;
    }
}

File: internal/content/testdata/bench/go/nc-rotting-oranges.go
func oranges_rotting(grid [][]int) int {
	m, n := len(grid), len(grid[0])
	type point struct{ i, j, t int }
	var queue []point
	fresh := 0
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j] == 2 {
				queue = append(queue, point{i, j, 0})
			} else if grid[i][j] == 1 {
				fresh++
			}
		}
	}
	if fresh == 0 {
		return 0
	}
	minutes := 0
	dirs := [][2]int{{1, 0}, {-1, 0}, {0, 1}, {0, -1}}
	for len(queue) > 0 {
		cur := queue[0]
		queue = queue[1:]
		for _, d := range dirs {
			ni, nj := cur.i+d[0], cur.j+d[1]
			if ni >= 0 && ni < m && nj >= 0 && nj < n && grid[ni][nj] == 1 {
				grid[ni][nj] = 2
				fresh--
				if cur.t+1 > minutes {
					minutes = cur.t + 1
				}
				queue = append(queue, point{ni, nj, cur.t + 1})
			}
		}
	}
	if fresh == 0 {
		return minutes
	}
	return -1
}
